pgmpy/factors/Factor.py

```diff
@@ -780,7 +780,9 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
     def __hash__(self):
-        variables = sorted(self.variables)
+        var_with_hashes = list(map((lambda var: (hash(var), var)), self.variables))
```

Comment:
@abhi95 No need to explicitly convert it to `list`. `sorted` can handle `map` objects.